src/v2/client/sessionStorageHelper.js

```diff
@@ -0,0 +1,29 @@
+/*!
+ * Copyright (c) 2021, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+const STATE_HANDLE_SESSION_STORAGE_KEY = 'osw-oie-state-handle';
+
+const removeStateHandle = () => {
+  sessionStorage.removeItem(STATE_HANDLE_SESSION_STORAGE_KEY);
+};
+const setStateHandle = (token) => {
+  sessionStorage.setItem(STATE_HANDLE_SESSION_STORAGE_KEY, token);
```

Comment:
This will be problematic with [Incognito Browsers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API#private_browsing_incognito_modes), as `setItem` is not available. We should consider a fallback if we're operating within the context of Private/Incognito browser modes.

Reply:
you are saying this will fail in private browser.. `sessionStorage.setItem('sdf','sdf')` ?

Also, new private browser should always starts with clear state and we don't persist storage across tabs.

Reply:
Good point! Are there any alternatives?

Reply:
I tried in chrome incognito mode, `setItem` does work but will be wiped out when close the incognito window.

Reply:
This issue is specifically for Safari:

> Then there are some browsers, most notably Safari, that have opted for a solution where storage is available, but is empty and has a quota of 0 bytes assigned, effectively making it impossible to write data to it.

One solution could be to fallback to a short-lived cookie, but relying on persisting the token in `sessionStorage` in Safari unfortunately will not work.

Reply:
Chatted offline and this would becomes limitation for now (cannot support page refresh at any  Browsers that disabled session storage)